packages/block-editor/src/components/link-control/index.js

```diff
@@ -78,8 +80,8 @@ function LinkControl( {
 
 		// Populate input searcher whether
 		// the current link has a title.
-		if ( value && value.title && mode === 'edit' ) {
-			setInputValue( value.title );
+		if ( value && value.url && mode === 'edit' ) {
+			setInputValue( value.url );
```

Comment:
I went back to url here @getdave since it's the most common denominator between all links.

Reply:
I see. I see why you are doing this, but do you think we should document an Issue regarding re-population of the text box with the most appropriate thing for a given link depending on it's `type`?

As mentioned previously, if we search for a `Post` and then select one of the search results, the resulting URL is of little concern to us. Therefore imagine my surprise as a user when having click "change" I'm presented with the URL of my `Post` rather than the original search text I entered.

Reply:
If we do what you suggest, what happens when you click "edit". I expect three things:

 - the "type" of the link need to be persisted in value/onChange to know how to apply it back
 - if I edit the link, I expect the selected post to be highlighted in the suggestions and I think the  text/title is not enough to perform this as it's ambiguous 
 - If I edit the link, I should be able to submit it again without any change,  keeping the exact same value, which doesn't seem possible today as well.

For these reasons, I think it's probably better to start with the "URL" (more deterministic so easier for resubmits, searches), we can explore an alternative though.

Reply:
Yes sorry in this PR I think URl is the best all-round solution. 

I'm mainly highlighting the potential UX issue here. I still feel it's relevant despite the technical and design hurdles. I'll raise an Issue unless you object?